guides/use-microprofile-lra.adoc

```diff
@@ -0,0 +1,165 @@
+
+We provide only a high-level overview of the LRA protocol in this post. The full overview of the protocol is available
+at https://download.eclipse.org/microprofile/microprofile-lra-2.0/microprofile-lra-spec-2.0.html.
+
+In LRA, the specification API utilizes annotations from the `org.eclipse.microprofile.lra.annotation` package. The main
+annotation is the `@LRA` which controls the life cycle of the LRA. It's use might seem similar to the use of
+`@Transactional` annotation from JTA, however, the transaction characteristics differ greatly. If you are interested in
+the comparison of the saga pattern to the ACID transactions, you can find an explanation in this talk from DevoxxUK -
+https://www.youtube.com/watch?v=7DI4xXv1xGU.
+
+The Narayana implementation utilizes the coordinator orchestration of the LRAs. The LRA coordinator is a standalone
+service that is responsible for the management operations of the LRAs started in the system. When any LRA participant
+(user service) wants to start a new LRA, it contacts the LRA coordinator that in turn returns the LRA ID of the newly
+started LRA that can be propagated by the LRA participant to any other services. When an LRA-aware service receives the
+LRA ID, it can optionally enlist within the same LRA which is again done by the enlistment call to the coordinator. When
+the LRA finishes (success or failure), the LRA coordinator is responsible for invocations of the completions or the
+compensations callbacks of all enlisted LRA participants.
+
+== Enabling MicroProfile LRA subsystems
+
+The integration of the LRA specification is included in two separate subsystems:
+
+- `microprofile-lra-coordinator` - The LRA coordinator responsible for starting, managing, and recovery of the LRAs.
+
+- `microprofile-lra-participant` - The client library utilized in user deployments to participate in the distributed LRAs and define compensation and completition callbacks.
+
+=== Required extensions and subsystems configuration
+
+The LRA extensions are not included in the standard configurations included with WildFly application server. They need
+to be explitcly enabled either in the configuration XML or by using CLI operations:
+
+[source,bash]
+----
+[standalone@localhost:9990 /] /extension=org.wildfly.extension.microprofile.lra-coordinator:add
+{"outcome" => "success"}
+
+[standalone@localhost:9990 /] /subsystem=microprofile-lra-coordinator:add
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+
+[standalone@localhost:9990 /] /extension=org.wildfly.extension.microprofile.lra-participant:add
+{"outcome" => "success"}
+
+[standalone@localhost:9990 /] /subsystem=microprofile-lra-participant:add
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+
+[standalone@localhost:9990 /] reload
+----
```

Comment:
I think we need to come up with a consistent style for describing provisioning / configuration operations in these guides. Think through options and only allow a limited set of permutations.

1) Initial state:

a) assume a complete install
b) assume a slimmed install has the prerequisites (modules are present, config has the needed basic resources)
c) assume no installation

2) Config updates

a) assume no install at all and describe provisioning the layers with the wildfly-maven-plugin
b) describe how to use the wildfly-maven-plugin to run CLI ops if needed
c) other means of describing CLI use. No assuming it's all understood, unless a prerequisite describes it.

This one is a combination of 1a or 1b and 2c, but with an assumption of CLI use. That's ok for a blog but I don't think it's great for a user guide.

The 'application-logging one was 1a/1b and 2b, which was nice. The 1b assumption works because most any reasonable slimmed provisioning would have the logging subsystem.